crates/universal-sierra-compiler-api/src/command.rs

```diff
@@ -0,0 +1,84 @@
+use anyhow::Context;
+use std::ffi::{OsStr, OsString};
+use std::path::PathBuf;
+use std::process::{Command, Stdio};
+
+/// A builder for `universal-sierra-compiler` command invocation.
+#[derive(Clone, Debug, Default)]
+#[allow(clippy::struct_excessive_bools)]
+pub struct UniversalSierraCompilerCommand {
+    args: Vec<OsString>,
+    current_dir: Option<PathBuf>,
+    inherit_stderr: bool,
+}
+
+impl UniversalSierraCompilerCommand {
+    /// Creates a default `universal-sierra-compiler` command,
+    /// which will look for `universal-sierra-compiler` in `$PATH`
+    #[must_use]
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Ensures that `universal-sierra-compiler` binary is available in the system.
+    pub fn ensure_available() -> anyhow::Result<()> {
```

Comment:
we should probably allow setting this with an .env variable as well

Reply:
Why? I think it's reasonable to expect a user to have a compiler in their PATH